src/memex/views.py

```diff
@@ -158,6 +158,7 @@ def create(request):
     schema = schemas.CreateAnnotationSchema(request)
     appstruct = schema.validate(_json_payload(request))
     annotation = storage.create_annotation(request, appstruct)
+    request.db.flush()
```

Comment:
So, we used to add the annotation object to the session before we updated the document meta data, which meant that it was automatically being flushed to the DB and thus it would have an id that is not `None`. But since that's not happening anymore (see my comment above), the annotation isn't flushed at this point (there's no need for it at this point, not like for the document related records which we need to query inside `update_document_metadata`). 
We then emit the created event, and [the reply notification subscribers tries to load the annotation from the database](https://github.com/hypothesis/h/blob/master/h/subscribers.py#L47) but it does have a `NULL` id at this point which then blows up.
Would it be nicer to move the flush into the `create_annotation` function? Doesn't make much sense to me since we don't care about it being flushed inside that function or any functions that we call from `create_annotation` (that's how update works for example).